### ISSUE TITLE
UpgradeSpringBoot should add ehcache version as it used to be part of the Spring BOM.

### ISSUE THREAD

**Issue by Jenson3210** [openrewrite/rewrite-spring#376]
## What version of OpenRewrite are you using?
I am using Gradle plugin 6.0.0

## How are you running OpenRewrite? 

- Gradle plugin
- [a7439c7 commit of spring-petclinic](https://github.com/spring-projects/spring-petclinic/tree/a7439c74ea718c4f59fe6c7c643c4afe59d7e718) as this is the first commit with gradle on petclinic repo

Can you share your configuration so that we can rule out any configuration issues? 

- gradle 7.4
- MAC OS

### Adapt build gradle to run rewrite: 
```groovy
plugins {
  id 'org.springframework.boot' version '2.6.1'
  id 'io.spring.dependency-management' version '1.0.11.RELEASE'
  id 'java'
  id 'org.openrewrite.rewrite' version '6.0.0'
}

rewrite {
    activeRecipe(
            'org.openrewrite.java.spring.boot3.UpgradeSpringBoot_3_0',
    )
}

//...
dependencies {
  rewrite platform('org.openrewrite.recipe:rewrite-recipe-bom:2.0.0')
  rewrite('org.openrewrite.recipe:rewrite-spring')
  ...
}
```

### Run openrewrite

Launch `gradle rewriteRun`. 
Wait for build succeeded message.

**Expected behaviour**

`ehcache ` dependency gets changed to `org.ehcache:ehcache:3.7.1` as the BOM does not add the version anymore.

**Actual behaviour**

It is not changed.

## Are you interested in [contributing a fix to OpenRewrite](https://github.com/openrewrite/.github/blob/main/CONTRIBUTING.md#contributing-fixes)?
Sure, if guided properly I am always open to take "newbie" tickets!

**Comment by timtebeek** [comment]
Thanks for opening this one as well @Jenson3210 ; as [noted](https://github.com/openrewrite/rewrite-spring/issues/349#issuecomment-1587288991) 

> We might need similar handling as we already have for `mysql-connector-j` applied to `ehcache`, as that also saw it's version number no longer managed, as you've pointed out. There might be more to it (adopting an alternative), but at least adding the version number should restore functionality post-migration for now.

https://github.com/openrewrite/rewrite-spring/blob/f0151affbff68d4c6538540ae4fe605ecabc0c2e/src/main/resources/META-INF/rewrite/spring-boot-27.yml#L33-L38

https://github.com/openrewrite/rewrite-spring/blob/f0151affbff68d4c6538540ae4fe605ecabc0c2e/src/main/resources/META-INF/rewrite/spring-boot-27.yml#L54-L59

**Comment by Jenson3210** [comment]
@timtebeek  as mentioned, I will try to find some time for this one as it appears trivial indeed!
Question: Do you know by any chance how to identify the spring-boot-{X} version to add the condition to?

**Comment by timtebeek** [comment]
> Do you know by any chance how to identify the spring-boot-{X} version to add the condition to?

It looks like they [removed ehcache 2.x , and then added ehcache 3.x](https://github.com/spring-projects/spring-boot/releases?q=ehcache&expanded=true) when going to Spring Boot 3; so likely this would fall under the Spring Boot 3.0 upgrade recipe.

**Comment by timtebeek** [comment]
Don't want to steal this issue from you if you're still planning to pick it up, but do kind of want to see this through. Any objections on us picking this up to fix my migration next demo?

**Comment by Jenson3210** [comment]
Totally lost track of this one! Apologies! 
No objections whatsoever. I will benefit if it makes it, and I do not have the time, nor the knowledge to fully implement this I think... 

I remember that I stumbled upon [this link](https://www.ehcache.org/documentation/3.3/migration-guide.html) that scared me off a bit to just make the PR 2 weeks ago. Forgot to ping here and then forgot about the issue.

So for consumers using the JCache api, there might not be any issues... But for migrators with ehcache config as xml or as code, there might be impact in code also. 
The version upgrade seemed easy, but the code rewriting did not.

**Comment by timtebeek** [comment]
Thanks for reporting back here! I think we can optimistically at least update the managed dependency; then if people chime in with issues pick those up separately. Right now things already break in the majority of cases due to the renamed dependency; then they'd only break in a subset of cases. Small steps.

@AlekSimpson could this be one for you to pick up? I'd expect a similar handling for ehache as we [already do for mysql-connector](https://github.com/openrewrite/rewrite-spring/issues/376#issuecomment-1587331281).

**Comment by Jenson3210** [comment]
@timtebeek 
I did an initial push of a fix but I am doing something wrong. [Fork branch](https://github.com/Jenson3210/rewrite-spring/tree/issue_376)

I am having a `java.lang.NoClassDefFoundError: javax/xml/bind/annotation/XmlElement`
But not doing anything different from the mysql tests from last time. 
The only thing I can think of is different dependencies being present as for the spring boot 3 classpath I would need jakarta xml? 
Not sure what the fix would be. 
Also, I think that with the latest push on main branch, the plugins will be adapted (so should be changed to other version also) but not sure as I cannot run. 

Also, the main is currently in failed state.

**Comment by timtebeek** [comment]
Feel free to start a draft pull request! That way we can push changes there as well to continue where you started. :)
We'll then also fix the issues on the main branch, and update your branch to get those fixes in.

**Comment by Jenson3210** [comment]
Here you go!

**Comment by AlekSimpson** [comment]
> Thanks for reporting back here! I think we can optimistically at least update the managed dependency; then if people chime in with issues pick those up separately. Right now things already break in the majority of cases due to the renamed dependency; then they'd only break in a subset of cases. Small steps.
> 
> @AlekSimpson could this be one for you to pick up? I'd expect a similar handling for ehache as we [already do for mysql-connector](https://github.com/openrewrite/rewrite-spring/issues/376#issuecomment-1587331281).

Ya I could probably take a look at this soon.

**Comment by timtebeek** [comment]
It looks like Spring Boot again manages a version for Ehcache, now at 3.10.8 in Spring Boot 3.1.3
https://docs.spring.io/spring-boot/docs/current/reference/html/dependency-versions.html
That removes the need for special handling, as users are likely to upgrade directly to 3.1 or 3.2, as support for 3.0 expires this November.

Thanks again for reporting this issue and the attempts to fix it in #383; I don't think we require anything to be done at this point.
